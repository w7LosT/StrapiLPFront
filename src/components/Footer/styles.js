import styled, { css } from "styled-components";
import {Text} from '../TextComponent/styles';
import {Container as SectionComponent} from '../SectionComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    border-top: 1px solid rgba(0, 0, 0, .3);

    & a{
      color: inherit;
      text-decoration: none;
    }

    & ${Text}{
      font-size: ${theme.fonts.sizes.sm};
    }
  `}
`;
