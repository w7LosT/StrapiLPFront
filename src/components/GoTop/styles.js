import styled, { css } from "styled-components";

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    bottom: 30px;
    right: 15px;
    width: 50px;
    height: 50px;
    font-size: ${theme.fonts.sizes.lg};
    color: ${theme.default.darkBG};
    background-color: ${theme.default.colorWhite};
    border-radius: 100%;
    border: 2px solid ${theme.default.darkBG};
    box-shadow: 0 3px 6px rgba(0, 0, 0, .4);
    z-index: 6;
    opacity: .7;
    transition: all .3s ease-in-out;

    &:hover{
      opacity: 1;
    }
  `}
`;
