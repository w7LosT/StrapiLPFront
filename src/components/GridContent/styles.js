import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme, background }) => css`
    max-width: 580px;
    margin: 0 auto;
    text-align: center;

    & h1{
      background: none;
      color: ${ background ? theme.default.colorWhite : theme.default.primaryColor };
    }

    & a{
      color: ${ background ? theme.default.colorWhite : theme.default.primaryColor };
    }
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.lg} 0;
  `}
`;
