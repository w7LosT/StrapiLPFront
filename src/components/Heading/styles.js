import styled, { css } from "styled-components";

const titleSize = {
  sm: (theme) => css`
    font-size: ${theme.fonts.sizes.sm};
  `,
  md: (theme) => css`
    font-size: ${theme.fonts.sizes.md};
  `,
  lg: (theme) => css`
    font-size: ${theme.fonts.sizes.lg};
    ${mediaFont(theme)};
  `,
};

const titleCaseFunction = {
  none: (theme) => css`
    text-transform: ${theme.fonts.cases.normal}
  `,
  uppercase: (theme) => css`
    text-transform: ${theme.fonts.cases.uppercase}
  `,
  lowercase: (theme) => css`
    text-transform: ${theme.fonts.cases.lowercase}
  `,
  capitalize: (theme) => css`
    text-transform: ${theme.fonts.cases.capitalize}
  `,
}

const mediaFont = (theme) => css`
  @media ${theme.medias.lteMedium}{
    font-size: ${theme.fonts.sizes.md};
  }
`;

export const Title = styled.h1`
  ${
    ({ theme, colorDark, size, titleCase }) => css `
      color: ${ colorDark ? theme.default.colorWhite : theme.default.mainColor };
      /*background-color: ${colorDark ? theme.default.secondaryBG : theme.default.secondaryBG};*/
      ${titleSize[size](theme)};
      ${titleCaseFunction[titleCase](theme)};
      min-width: 200px;

      > a img{
        margin: 0 auto;
      }

      @media ${theme.medias.lteMedium}{
        margin: 15px 0;
      }
    `
  }
`;
