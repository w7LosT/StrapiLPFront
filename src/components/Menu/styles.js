import styled, { css } from "styled-components";
import { Container as SectionComponent } from '../SectionComponent/styles';

export const isMenuOpen = (theme) => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, visible }) => css`
    position: fixed;
    top: 0;
    left: 0;
    width: 100vw;
    display: flex;
    align-items: center;
    justify-content: space-between;
    border-bottom: 2px solid ${theme.default.mainBG};
    background-color: ${theme.default.colorWhite};
    z-index: 5;
    transition: all .3s ease;

    & ${SectionComponent}{
      min-width: 800px;
      padding-top: 12px;
      padding-bottom: 12px;
    }

    @media ${theme.medias.lteMedium}{
      height: 100vh;
      visibility: hidden;
      opacity: 0;
      ${visible && isMenuOpen(theme)};

      & ${SectionComponent}{
        min-width: 100%;
        overflow-y: auto;
      }
    }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;
    align-items: center;
    justify-content: space-between;

    @media ${theme.medias.lteMedium}{
        flex-flow: column wrap;
        justify-content: center;
        height: 100%;
    }
  `}
`;

export const BurgerButton = styled.button`
  ${({ theme, visible }) => css`
    z-index: 6;
    position: fixed;
    top: 0;
    right: 0;
    padding: 5px;
    background-color: ${theme.default.darkBG};
    color: ${theme.default.colorWhite};
    border: none;
    cursor: pointer;
    display: none;
    pointer-events: ${visible ? 'none' : 'all'};

    > svg{
      width: 35px;
      height: 35px;
    }

    @media ${theme.medias.lteMedium}{
      display: flex;
      align-items: center;
      justify-content: center;
    }
  `}
`;
