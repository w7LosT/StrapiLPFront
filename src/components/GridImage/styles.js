import styled, { css } from "styled-components";
import { Title as  HeadingContainer } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    padding: 10px;
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.lg};
    padding: ${theme.spacings.xlg} 0;
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    transition: all .3s ease;
    box-shadow: 0 3px 6px rgba(0, 0, 0, .5);

    &:hover{
      transform: scale(1.1);
      box-shadow: 0 6px 8px rgba(0, 0, 0, .5);

    }
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;

  `}
`;
