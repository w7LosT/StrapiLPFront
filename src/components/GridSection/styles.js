import styled, { css } from "styled-components";
import { Title as  HeadingContainer } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: 1140px;
    margin: 0 auto;
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.lg};
    padding: ${theme.spacings.xlg} 0;
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer}{
      position: relative;
      left: 35px ;
      margin-bottom: 10px;
    }

    ${HeadingContainer}::before{
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.fonts.sizes.xlg};
      top: -1rem;
      left: -3rem;
      transform: rotate(5deg);
    }
  `}
`;
