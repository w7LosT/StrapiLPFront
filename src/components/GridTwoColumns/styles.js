import styled, { css } from "styled-components";
import {Title} from '../Heading/styles';

export const Container = styled.div`
  ${({ theme, background }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    gap: ${theme.spacings.lg};
    align-items: center;
    padding: 25px;

    ${Title}{
      margin-bottom: ${theme.spacings.lg};
      text-align: center;
      background: none !important;
      color: ${background ? theme.default.colorWhite : theme.default.mainColor};
    }

    @media ${theme.medias.lteMedium}{
      grid-template-columns: 1fr;
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css``}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    text-align: center;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    max-width: 600px;

    @media ${theme.medias.lteMedium}{
      max-width: 300px;
    }
  `}
`;
