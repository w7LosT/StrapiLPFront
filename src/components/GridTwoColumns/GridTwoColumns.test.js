import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { GridTwoColumns } from '.';

const mock = {
  title: 'GridTwoColumns',
  text: 'Lorem ipsum dolor, sit amet consectetur adipisicing elit. Dolores veniam ducimus praesentium dolore laborum hic. At, tenetur vitae necessitatibus pariatur corrupti praesentium sit perspiciatis, eveniet repellendus possimus dicta labore consequatur.',
  imgSrc: 'assets/images/html_css_js.svg',
  background: false,
}

describe('<GridTwoColumns />', () => {
  it('should render', () => {
    const {container} = renderTheme(<GridTwoColumns {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
