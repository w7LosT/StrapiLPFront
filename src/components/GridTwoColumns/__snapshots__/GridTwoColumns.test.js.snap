// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumns /> should render 1`] = `
<div>
  <div
    class="sc-jRQBWg fkdoeC"
  >
    <div
      class="sc-gsDKAQ iUXaVc"
    >
      <div
        class="sc-dkPtRN cKQdCP"
      >
        <h2
          class="sc-bdvvtL kIeIKU"
        >
          GridTwoColumns
        </h2>
        <div
          class="sc-gKclnd jqWrFg"
        >
          Lorem ipsum dolor, sit amet consectetur adipisicing elit. Dolores veniam ducimus praesentium dolore laborum hic. At, tenetur vitae necessitatibus pariatur corrupti praesentium sit perspiciatis, eveniet repellendus possimus dicta labore consequatur.
        </div>
      </div>
      <div
        class="sc-hKwDye chNkyf"
      >
        <img
          alt="Imagem"
          class="sc-eCImPb gcUvKC"
          src="assets/images/html_css_js.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
